### ISSUE TITLE
Boot2Docker: VOLUME prevents chown to have effect

### ISSUE THREAD

**Issue by ChristianKniep** [ipfs/go-ipfs#1804]
Hey there,
I like to tip my toe into ipfs, therefore I tried to create my own docker image - failed - and came to your very own dockerfile. :)
If I build and run the image, I get:
```
$ docker build --rm -t ipfs/go-ipfs .
$ docker run -ti -p 8080:8080 -p 4001:4001 -p 5001:5001 ipfs/go-ipfs:latest
ERR: /data/ipfs is not writable for user 'ipfs' (UID 1000)
```
Which is caused by the fact that `VOLUME /data/ipfs` is done under the user `root`. Which creates a directory on the docker host, which is mounted into the container using the give user.
By putting the line `VOLUME` underneath the `USER ipfs` line, it will work as expected. This might take effect due to newer docker version (I use 1.8).

**Comment by ChristianKniep** [comment]
#1805

**Comment by ChristianKniep** [comment]
Sorry, I tried to fix the signed-off check.. it's #1806 now...

**Comment by jbenet** [comment]
cc @Luzifer PTAL



—

**Comment by chriscool** [comment]
Fixed by #2064 (Docker image volume fix and Sharness tests).
Thanks @ChristianKniep !
